src/App.jsx

```diff
@@ -1,17 +1,44 @@
+import React, { useState } from 'react';
 import './App.css';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import chatMessages from './data/messages.json';
 
 const App = () => {
+  // const [entries, setEntries] = useState(chatMessages);
+  const [entries, setEntries] = useState(
+  chatMessages.map((msg) => ({ ...msg, liked: msg.liked ?? false }))
+);
```

Comment:
We can assume that all `msg`s in `chatMessages will have a property `liked` so we don't need to iterate each of them. We can directly set the initial state `entries` equal to `chatMessages`, like what you have commented out on line 8.

```suggestion
  const [entries, setEntries] = useState(chatMessages);
);
```